### ISSUE TITLE
Objectosphere is computed on the softmax prediction, but not on the embedding layer

### ISSUE THREAD

**Issue by siebenkopf** [Vastlab/Reducing-Network-Agnostophobia#4]
When I understand the loss function:
https://github.com/Vastlab/Reducing-Network-Agnostophobia/blob/581b3f06aeedb737e83a8610573f02299e2416b2/MNIST/Mnist_Training.py#L69
correctly, the input to your loss function are the one-hot-encoded target values (`y_true`) and the estimated softmax probabilities (`y_pred`). From this `y_pred`, you compute the magnitude `pred` for each sample. However, this magnitude should be computed from the output of the layer before the logits, not from the softmax probabilities.

This might be the reason, why there is very little difference between the Entropic Open-Set loss and the Objectosphere loss.

**Comment by siebenkopf** [comment]
Sorry, I should always try to read the whole thing before writing a bug report. In
https://github.com/Vastlab/Reducing-Network-Agnostophobia/blob/581b3f06aeedb737e83a8610573f02299e2416b2/MNIST/Mnist_Training.py#L199
your clearly apply this function on the `fc` layer, as expected.

You should consider renaming the `y_pred` parameter of the function call to `fc` or something alike to avoid confusion.
